### PR TITLE
#232329  smaller pr that only includes the stieName + folderPath

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
dotcms:

We're making CT Resource smarter to handle invalid folder/site names. Making it default to default-host/system-folder

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolver.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    String CT_SKIP_RESOLVE_SITE = "CT_SKIP_RESOLVE_SITE";
+
+    String CT_FALLBACK_DEFAULT_SITE = "CT_FALLBACK_DEFAULT_SITE";
+
+    ContentType resolveSiteAndFolder(ContentType contentType)
+            throws DotDataException, DotSecurityException;
+
+    static SiteAndFolderResolver newInstance(final User user) {
+        return new SiteAndFolderResolverImpl(user,
+                Config.getBooleanProperty(CT_SKIP_RESOLVE_SITE, false),
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolver.java:20]: if this is call so often, please use a Lazy. Config is a blocking code

File: dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolverImpl.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+     * The last one takes precedence over siteId
+     * meaning that if both are provided siteName will be used to resole
+     * if a site-name is passed and the resolution fails we fall back to default site
+     * if nothing is passed the site is resolved to SYSTEM_HOST
+     * @param params
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @NotNull
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolverImpl.java:98]: remove me

File: dotCMS/src/main/java/com/dotcms/contenttype/model/component/ResolvedSiteAndFolder.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.dotcms.contenttype.model.component;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@JsonSerialize(as = ImmutableResolvedSiteAndFolder.class)
+@JsonDeserialize(as = ImmutableResolvedSiteAndFolder.class)
+@Value.Immutable
+public interface ResolvedSiteAndFolder {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/component/ResolvedSiteAndFolder.java:10]: Doc me

File: dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolver.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolver.java]: done

**Review by jcastro-dotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**